src/imitation/rewards/reward_nets.py

```diff
@@ -409,6 +411,203 @@ def forward(self, state, action, next_state, done):
+    """
+
+    def __init__(
+        self,
+        observation_space: gym.Space,
+        action_space: gym.Space,
+        use_state: bool = True,
+        use_action: bool = True,
+        use_next_state: bool = False,
+        use_done: bool = False,
+        hwc_format: bool = True,
+        **kwargs,
+    ):
+        """Builds reward CNN.
+
+        Args:
+            observation_space: The observation space.
+            action_space: The action space.
+            use_state: Should the current state be included as an input to the CNN?
+            use_action: Should the current action be included as an input to the CNN?
+            use_next_state: Should the next state be included as an input to the CNN?
+            use_done: Should the "done" flag be included as an input to the CNN?
+            hwc_format: Are image inputs in (h,w,c) format (True), or (c,h,w) (False)?
+                If hwc_format is False, image inputs are not transposed.
+                Note that this affects both state and action inputs.
+            kwargs: Passed straight through to `build_cnn`.
+
+        Raises:
+            ValueError: if observation or action space is not easily massaged into a
+                CNN input.
+        """
+        super().__init__(observation_space, action_space)
+        self.use_state = use_state
+        self.use_action = use_action
+        self.use_next_state = use_next_state
+        self.use_done = use_done
+        self.input_size = 0
+        self.obs_is_image = preprocessing.is_image_space(observation_space)
+        self.act_is_box_with_channels = (
+            isinstance(action_space, spaces.Box) and len(action_space.shape) == 3
+        )
+        self.hwc_format = hwc_format
+
+        if not self.can_handle_space(observation_space):
+            raise ValueError(
+                "CnnRewardNet doesn't know how to deal with this observation space.",
+            )
+        if not self.can_handle_space(action_space):
+            raise ValueError(
+                "CnnRewardNet doesn't know how to deal with this action space.",
+            )
+
+        if self.use_state:
+            self.input_size += self.get_num_channels(observation_space)
+
+        if self.use_action:
+            self.input_size += self.get_num_channels(action_space)
+
+        if self.use_next_state:
+            self.input_size += self.get_num_channels(observation_space)
+
+        if self.use_done:
+            self.input_size += 1
+
+        full_build_cnn_kwargs = {"hid_channels": (32, 32)}
+        full_build_cnn_kwargs.update(kwargs)
+        full_build_cnn_kwargs.update(
+            {
+                "in_channels": self.input_size,
+                "out_size": 1,
+                "squeeze_output": True,
+            },
+        )
+
+        self.cnn = networks.build_cnn(**full_build_cnn_kwargs)
+
+    def can_handle_space(self, my_space: gym.Space) -> int:
```

Comment:
Given that at this point the convo is about `imitation`'s type-checking rather than this PR, I'd like to resolve this thread. Will give you two some time to turn it into an issue or something before resolving it on Monday.

Reply:
I'll add this as an issue. We can have mypy as suggestions for now, and not enforce it until we customize what errors it spits out.